google/cloud/security/inventory/pipelines/load_appengine_pipeline.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Pipeline to load appengine applications into Inventory.
+"""
+
+from google.cloud.security.common.data_access import project_dao as proj_dao
+from google.cloud.security.common.util import log_util
+from google.cloud.security.common.util import parser
+from google.cloud.security.inventory.pipelines import base_pipeline
+
+LOGGER = log_util.get_logger(__name__)
+
+
+class LoadAppenginePipeline(base_pipeline.BasePipeline):
+    """Load all AppEngine applications for all projects."""
+
+    RESOURCE_NAME = 'appengine'
+
+    def _retrieve(self):
+        """Retrieve AppEngine applications from GCP.
+
+        Get all the projects in the current snapshot and retrieve the
+        AppEngine applications for each.
+
+        Returns:
+            A dict mapping projects with their AppEngine applications:
+            {project_id: application}
+        """
+        projects = proj_dao.ProjectDao().get_projects(self.cycle_timestamp)
+        apps = {}
+        for project in projects:
+            app = self.api_client.get_app(project.id)
+            if app:
+                apps[project.id] = app
+        return apps
+
+    def _transform(self, resource_from_api):
+        """Create an iterator of AppEngine applications to load into database.
+
+        Args:
+            resource_from_api: A dict of AppEngine applications, keyed by
+                project id, from GCP API.
+
+        Yields:
+            Iterator of AppEngine applications in a dict.
+        """
+        for project_id, app in resource_from_api.iteritems():
+            yield {'project_id': project_id,
+                   'name': app.get('name'),
+                   'app_id': app.get('id'),
+                   'dispatch_rules': parser.json_stringify(
+                       app.get('dispatchRules', [])),
+                   'auth_domain': app.get('authDomain'),
+                   'location_id': app.get('locationId'),
+                   'code_bucket': app.get('codeBucket'),
+                   'default_cookie_expiration': app.get(
+                       'defaultCookieExpiration'),
+                   'serving_status': app.get('servingStatus'),
+                   'default_hostname': app.get('defaultHostname'),
+                   'default_bucket': app.get('defaultBucket'),
+                   'iap': parser.json_stringify(app.get('iap', {})),
+                   'gcr_domain': app.get('gcrDomain')}
```

Comment:
I know we haven't been super consistent (until last week) about this, but can you add the "raw_appengine_app" json data to the appengine table? Henry created a PR for an example:

https://github.com/GoogleCloudPlatform/forseti-security/pull/339/files